packages/grid/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx

```diff
@@ -2654,4 +2654,28 @@ describe('<DataGridPremium /> - Row Grouping', () => {
     });
     await waitFor(() => expect(getCell(1, 3).textContent).to.equal('username 2'));
   });
+
+  // See https://github.com/mui/mui-x/issues/8853
+  it('should not reorder rows after calling `updateRows`', async () => {
+    render(
+      <Test
+        columns={[{ field: 'id' }, { field: 'group' }, { field: 'username', width: 150 }]}
+        rows={[
+          { id: 1, group: 'A', username: 'username1' },
+          { id: 2, group: 'A', username: 'username2' },
+        ]}
+        rowGroupingModel={['group']}
+        defaultGroupingExpansionDepth={-1}
+      />,
+    );
+
+    expect(getColumnValues(3)).to.deep.equal(['', 'username1', 'username2']);
+
+    // trigger row update without any changes in row data
+    act(() => apiRef.current.updateRows([{ id: 1 }]));
```

Comment:
Should we also try an actual data update (lets say `username2` => `username2 updated`)

Reply:
There already is a test that covers an actual update:

https://github.com/mui/mui-x/blob/e78f8c5e76ea74b67faa3485f589dec6d34e142b/packages/grid/x-data-grid-premium/src/tests/rowGrouping.DataGridPremium.test.tsx#L2622C9-L2636

In the regression covered by the test in this PR, the row was reordered even if there were no actual updates to the row - that's why I wrote it this way 🙂